ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl.hs

```diff
@@ -224,7 +224,7 @@ openDBInternal args launchBgTasks = runWithTempRegistry $ do
             , getLedgerBackingStoreValueHandle = \rreg p -> getEnv h $ \env' -> do
                 Query.getLedgerBackingStoreValueHandle env' rreg p
 
-            , getBackingStore       =       getEnv h (pure . LgrDB.lgrBackingStore . cdbLgrDB)
+            , getLedgerTablesAtFor  = \pt keys -> getEnv h (LgrDB.getLedgerTablesAtFor pt keys . cdbLgrDB)
             , withLgrReadLock       = \m -> getEnv h (flip LgrDB.withReadLock m    . cdbLgrDB)
```

Comment:
Are we still using `withLgrReadLock`? If so we might have to check if and why such lock is necessary.

Reply:
Yes, the forging loop locks it. It must stay because we are supposed to continue the forging even if the chaindb selects a different chain after we acquired the tip and we actually forged, to comply with pre-utxo-hd behavior